apps/studio/src/server/modules/page/page.router.ts

```diff
@@ -7,18 +10,44 @@ import {
   updatePageSchema,
 } from "~/schemas/page"
 import { protectedProcedure, publicProcedure, router } from "~/server/trpc"
+import { safeJsonParse } from "~/utils/safeJsonParse"
 import {
   getFooter,
   getFullPageById,
   getNavBar,
+  getPageById,
   updateBlobById,
   updatePageById,
 } from "../resource/resource.service"
 import { getSiteConfig } from "../site/site.service"
 
+const typeCompiler = TypeCompiler.Compile(IsomerPageSchema)
+
 // TODO: Need to do validation like checking for existence of the page
 // and whether the user has write-access to said page
 const pageProcedure = protectedProcedure
+const validatedPageProcedure = pageProcedure.use(async ({ next, rawInput }) => {
+  if (
+    typeof rawInput === "object" &&
+    rawInput != null &&
+    "content" in rawInput
+  ) {
+    // NOTE: content will be the entire page schema for now...
+    if (!typeCompiler.Check(JSON.parse(rawInput.content as string))) {
+      throw new TRPCError({
+        code: "BAD_REQUEST",
+        message: "Schema validation failed.",
```

Comment:
do we want to return the exact error for better dev experience (and error debugging when something is thrown)

Reply:
Returning the error iterable as cause! Or is there a better way to do it here?